dev/run-tests-jenkins.py

```diff
@@ -0,0 +1,253 @@
+                                  '--write-out', http_code_header + '%{http_code}',
+                                  '--header', 'Content-Type: application/json',
+                                  comments_url],
+                                 stdout=subprocess.PIPE,
+                                 stderr=subprocess.PIPE)
+    curl_stdout, curl_stderr = curl_proc.communicate()
+    curl_returncode = curl_proc.returncode
+    # find all lines relevant to the Github API response
+    api_response = "\n".join([l for l in curl_stdout.split('\n')
+                              if l and not l.startswith(http_code_header)])
+    # find the line where `http_code_header` exists, split on ':' to get the
+    # HTTP response code, and cast to an int
+    http_code = int(curl_stdout[curl_stdout.find(http_code_header):].split(':')[1])
+
+    if not curl_returncode == 0:
+        print_err("Failed to post message to GitHub.")
+        print_err(" > curl_status:", curl_returncode)
+        print_err(" > curl_output:", curl_stdout)
+        print_err(" > data:", posted_message)
+
+    if http_code and not http_code == 201:
+        print_err(" > http_code:", http_code)
+        print_err(" > api_response:", api_response)
+        print_err(" > data:", posted_message)
+
+    if curl_returncode == 0 and http_code == 201:
+        print(" > Post successful.")
+
+
+def send_archived_logs():
+    print("Archiving unit tests logs...")
+
+    log_files = run_cmd(['find', '.',
+                         '-name', 'unit-tests.log',
+                         '-o', '-path', './sql/hive/target/HiveCompatibilitySuite.failed',
+                         '-o', '-path', './sql/hive/target/HiveCompatibilitySuite.hiveFailed',
+                         '-o', '-path', './sql/hive/target/HiveCompatibilitySuite.wrong'],
+                        return_output=True)
+
+    if log_files:
+        log_archive = "unit-tests-logs.tar.gz"
+
+        run_cmd(['tar', 'czf', log-archive] + log_files.strip().split('\n'))
+
+        jenkins_build_dir = os.environ["JENKINS_HOME"] + "/jobs/" + os.environ["JOB_NAME"] + \
+            "/builds/" + os.environ["BUILD_NUMBER"]
+
+        scp_proc = subprocess.Popen(['scp', log_archive,
+                                     'amp-jenkins-master:' + jenkins_build_dir + '/' + log_archive],
+                                    stdout=subprocess.PIPE,
+                                    stderr=subprocess.PIPE)
+        scp_stdout, scp_stderr = scp_proc.communicate()
+        scp_returncode = scp_proc.returncode
+
+        if not scp_returncode == 0:
+            print_err("Failed to send archived unit tests logs to Jenkins master.")
+            print_err(" > scp_status:",  scp_returncode)
+            print_err(" > scp_output:", scp_stdout)
+        else:
+            print(" > Send successful.")
+    else:
+        print_err(" > No log files found.")
+
+        rm_r(log_archive)
+
+
+def run_pr_tests(pr_tests, ghprb_actual_commit, sha1):
```

Comment:
This name is potentially confusing to new readers who aren't familiar with our existing terminology for this script, since it sounds like this might be running unit tests when it's actually running style / linting / dependency / compatibility checks.  Maybe naming it something like `run_pr_checks` and adding a comment would help to disambiguate.